src/main/java/practice/StreamPractice.java

```diff
@@ -23,7 +33,12 @@ public int findMinEvenNumber(List<String> numbers) {
      * But before that subtract 1 from each element on an odd position (having the odd index).
      */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        return IntStream.range(0, numbers.size())
+                .mapToObj(index -> (index % 2 == 0) ? numbers.get(index) : numbers.get(index) - 1)
```

Comment:
replace `mapToObj` with `map` and then you can remove `.mapToDouble(integer -> integer)`
```suggestion
                .mapToObj(index -> index % 2 == 0 ? numbers.get(index) : numbers.get(index) - 1)
```